### PR TITLE
[Core] Exposing MPI serializer constructor with pickle

### PR DESCRIPTION
**Pull request by marcnunezc**
Hi,

In this PR we are adding the pickle constructor missing in the MPI serializer python object.

### CODE REVIEW

**Comment by philbucher** [comment]
could you please add a test (if it makes sens) to see how this is being used?

**Comment by marcnunezc** [comment]
Sure, will add one soon!

**Comment by marcnunezc** [comment]
@philbucher I added a test in `Kratos/kratos/mpi/tests`. Let me know if you consider that this is the right place.

The idea is that this could be run in distributed environtments using, for instance, pyComps:

https://github.com/KratosMultiphysics/Kratos/wiki/How-to-run-multiple-cases-using-PyCOMPSs

Here we are testing that we can correctly serialize each partition of the mesh, packing it using pickle and send it to a "task". Each defined "task" (which has to be a function, cant be the method of a class) is what PyCompss sends to every processor available. 

If we did several calls to the executeComputeArea_Task() using pyCompss, each of the instances of the executeComputeArea_Task() calls would be computed in different computers in parallel.

File: kratos/mpi/tests/test_mpi_serializer.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+        current_model = KratosMultiphysics.Model()
+        model_part_to_read = current_model.CreateModelPart("ModelPart")
+        model_part_to_read.AddNodalSolutionStepVariable(KratosMultiphysics.PARTITION_INDEX)
+
+        importer_settings = KratosMultiphysics.Parameters("""{
+            "model_import_settings": {
+                "input_type": "mdpa",
+                "input_filename": "test_mpi_serializer",
+                "partition_in_memory" : false
```
Comment (philbucher) [inline kratos/mpi/tests/test_mpi_serializer.py]: why not?
Then you don't have to remove the partitioned files afterwards

I use it in my tests and works fine

File: kratos/mpi/tests/test_mpi_serializer.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+            },
+            "echo_level" : 0
+        }""")
+
+        model_part_import_util = distributed_import_model_part_utility.DistributedImportModelPartUtility(model_part_to_read, importer_settings)
```
Comment (philbucher) [inline kratos/mpi/tests/test_mpi_serializer.py:86]: this needs metis for partitioning => requires a skiptest

File: kratos/mpi/tests/test_mpi_serializer.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        serialized_model = KratosMultiphysics.MpiSerializer()
+        serialized_model.Save("ModelSerialization",current_model)
+
+        self.pickled_data = pickle.dumps(serialized_model, protocol=2) # Second argument is the protocol and is NECESSARY (according to pybind11 docs)
+
+    def tearDown(self):
+        communicator = KratosMultiphysics.DataCommunicator.GetDefault()
+        rank = communicator.Rank()
+        if rank == 0:
+            kratos_utilities.DeleteFileIfExisting("test_mpi_serializer.time")
```
Comment (philbucher) [inline kratos/mpi/tests/test_mpi_serializer.py]: this is safe to be called on all ranks

File: kratos/mpi/tests/test_mpi_serializer.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+        kratos_utilities.DeleteFileIfExisting("test_mpi_serializer_"+str(rank)+".mdpa")
+        kratos_utilities.DeleteFileIfExisting("test_mpi_serializer_"+str(rank)+".time")
+        communicator.Barrier()
```
Comment (philbucher) [inline kratos/mpi/tests/test_mpi_serializer.py]:
```suggestion
```

better not use barrier unless you need it

**Review by philbucher — COMMENTED** [review]
looks good to me, minor comments on the test
and can you please move the mdpa file to a separate folder like it is done with the regular tests in the core (=> `auxiliar_files_for_python_unnitest`)?
Thx

approval I leave to the others since I am not the expert here

File: kratos/mpi/tests/test_mpi_serializer.py (hunk 1 above)
Comment (rubenzorrilla) [reply kratos/mpi/tests/test_mpi_serializer.py]: Aside of the fact that it is much faster.

**Comment by marcnunezc** [comment]
All comments have been adressed!

File: kratos/mpi/tests/test_mpi_serializer.py [hunk 5]
```diff
@@ -0,0 +1,111 @@
+import KratosMultiphysics
+import KratosMultiphysics.mpi as KratosMPI
+import KratosMultiphysics.kratos_utilities as kratos_utilities
+import KratosMultiphysics.KratosUnittest as UnitTest
+import sys
+import os
+
+dependencies_are_available = kratos_utilities.CheckIfApplicationsAvailable("MetisApplication")
+if dependencies_are_available:
```
Comment (philbucher) [inline kratos/mpi/tests/test_mpi_serializer.py]: `distributed_import_model_part_utility` is safe to import but will fail if you use metis
```suggestion

```

Comment (marcnunezc) [reply kratos/mpi/tests/test_mpi_serializer.py]: You are right!

**Comment by philbucher** [comment]
looks good to me :+1:

**Review by RiccardoRossi — APPROVED** [review]

**Pull request merged**